earthfile2llb/converter.go

```diff
@@ -2114,20 +2128,32 @@ func (c *Converter) joinRefs(relRef domain.Reference) (domain.Reference, error)
 }
 
 func (c *Converter) checkAllowed(command cmdType) error {
+	if command == projectCmd && !c.ftrs.UseProjectSecrets {
+		return errors.New("--use-project-secrets must be enabled in order to use PROJECT")
+	}
+
+	if (command == pipelineCmd || command == triggerCmd) && !c.ftrs.UsePipelines {
+		return errors.New("--use-pipelines must be enabled in order to use PIPELINE or TRIGGER")
+	}
+
 	if c.mts.Final.RanInteractive && !(command == saveImageCmd || command == saveArtifactCmd) {
 		return errors.New("If present, a single --interactive command must be the last command in a target")
 	}
 
-	if c.mts.Final.RanFromLike {
-		return nil
+	if command == pipelineCmd && c.isPipeline {
+		return errors.New("only 1 PIPELINE command is allowed")
 	}
 
-	switch command {
-	case fromCmd, fromDockerfileCmd, locallyCmd, buildCmd, argCmd, importCmd:
-		return nil
-	default:
-		return errors.New("the first command has to be FROM, FROM DOCKERFILE, LOCALLY, ARG, BUILD or IMPORT")
+	if !c.mts.Final.RanFromLike {
+		switch command {
+		case fromCmd, fromDockerfileCmd, locallyCmd, buildCmd, argCmd, importCmd, projectCmd:
```

Comment:
I would make `PIPELINE` a from-like command and thus add it to this list? Nothing seems to break if this is not classified as such so far (since ARG and BUILD are allowed without a FROM too), but it feels like the logical thing to do in case of possible future expansion.

Reply:
Sure, makes sense.